pymedphys/_dicom/anonymise.py

```diff
@@ -642,9 +619,50 @@ def _filter_identifying_keywords(keywords_to_leave_unchanged):
     return keywords_filtered
 
 
-def get_anonymous_replacement_value(keyword):
-    """Get an appropriate dummy anonymisation value for a DICOM element
+def get_anonymous_replacement_value(
+    keyword,
+    current_value=None,
+    replacement_strategy=anon.ANONYMISATION_HARDCODE_DISPATCH,
```

Comment:
I really like this design. It's really nice. It makes my mind happy.

Reply:
On a separate point though, it is best to avoid mutable default parameters, see the following:

https://docs.python-guide.org/writing/gotchas/#what-you-should-do-instead

Reply:
I'm happy to get rid of the mutable default parameters, but that was a useful temporary mechanism for conveying the concept as well as exercising the pathway.  So I will change that soon

Reply:
The trick is to just make the default be None, and then have a if replacement_strategy is None --> replacement_strategy = dict

Functionally it is exactly the same, it has just side stepped the gothya.